ui/src/app/views/rename-modal/component.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+
+import {
+  Button
+} from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import {ReactComponent} from 'app/utils';
+
+
+interface RenameModalProps {
+  resource: {name: string};
+  onRename: Function;
+  onCancel: Function;
+}
+interface RenameModalState {
+  loading: boolean;
+  newName: string;
+}
+
+@ReactComponent({
+  selector: 'app-rename-modal',
+  propNames: ['resource', 'onRename', 'onCancel']
```

Comment:
It would be great to have a utility that read the key names from an interface and spat out an array of strings. I think I found some external libraries that do that? Or we could roll our own pretty easily. (To avoid the duplication here)

Reply:
Good suggestion, but apparently non-trivial in typescript with either the object type or interface type since they don't exist at runtime. https://github.com/Microsoft/TypeScript/issues/13267

There's apparently a custom typescript plugin which lets you do that, but looks questionable and probably not worth bringing in.

Something like this would also work ([per SO](https://stackoverflow.com/questions/45670705/typescript-iterate-interface-properties)), but probably adds more complexity than it saves in boilerplate - especially since this is transitional anyways:

```
const RenameModalProps = {
  resource: {name: ""};
  onRename: ()=>{};
  onCancel: ()=>{};
};
type RenameModalProps = typeof RenameModalProps;
...

{
  propNames: Object.keys(RenameModalProps)
```